File: javascript/puzzles/toInt.js
const toInt = s => {
    let i = 0;
    let n = 0;
    let size = s.length;

    while (i < size) {
        n = 10 * n + s.charCodeAt(i++) - '0'.charCodeAt(0);
    }

    return n;
};

console.log(toInt('7115'));

File: c/puzzles/toInt.c
#include <stdio.h>

int toInt(char *s) {
    int i, n, size;

    i = n = 0;

    for (size = 0; s[size] != '\0'; size++)
        ;

    while (i < size) {
        n = n * 10 + s[i] - '0';
        i++;
    }

    return n;
}

int main(int argc, char **argv) {
    int n = toInt("65");
    printf("%d %c\n", n, n);

    return 0;
}

